File: src/Model/Expression/VariableExpression.java
package Model.Expression;

import Model.Utils.IDictionary;
import Model.Utils.IHeap;

public class VariableExpression implements IExpression {
    private String id;

    public VariableExpression(String id) {
        this.id = id;
    }

    @Override
    public int eval(IDictionary<String, Integer> dict, IHeap<Integer, Integer> heap) {
        return dict.getValue(id);
    }

    @Override
    public String toString() {
        return id;
    }
}

File: Toy-Language-C#-Intepreter/Toy-Language-CS-Interpreter/Toy-Language-CS-Interpreter/Model/Expression/VariableExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language_CS_Interpreter.Model.Containers;

namespace Toy_Language_CS_Interpreter.Model.Expression
{
    class VariableExpression : IExpression
    {
        private string id;

        public VariableExpression(string id)
        {
            this.id = id;
        }

        public int Evaluate(Containers.IDictionary<string, int> symbolTable)
        {
            return symbolTable.Get(id);
        }

        public override string ToString()
        {
            return id;
        }
    }
}
